### ISSUE TITLE
Basic auth is not working

### ISSUE THREAD

**Issue by hasso** [jacksontj/promxy#70]
A simple config with basic auth configured leads to panic. There is no documentation regarding basic auth, but from the source it seems that it should work.

```
global:
  evaluation_interval: 5s

promxy:
  server_groups:
    - static_configs:
      - targets:
        - localhost/prometheus1
        - localhost/prometheus2
      anti_affinity: 10s
      http_client:
        basic_auth:
          username: 'user'
          password: 'password'
        tls_config:
          insecure_skip_verify: true
```
```
./promxy --config=promxy.yml
panic: interface conversion: http.RoundTripper is *config.basicAuthRoundTripper, not *http.Transport

goroutine 1 [running]:
github.com/jacksontj/promxy/servergroup.(*ServerGroup).ApplyConfig(0xc42029ad20, 0xc420362200, 0x1, 0xc4202048c8)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/servergroup/servergroup.go:133 +0x40e
github.com/jacksontj/promxy/proxystorage.(*ProxyStorage).ApplyConfig(0xc420539b70, 0xc4204601c0, 0xc4204601c0, 0x0)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/proxystorage/proxy.go:70 +0x146
main.reloadConfig(0xc420540180, 0x5, 0x8, 0x1d36266, 0x8)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/cmd/promxy/main.go:84 +0xc4
main.main()
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/cmd/promxy/main.go:303 +0x13a7
```

**Comment by jacksontj** [comment]
Thanks for the report! I have a fix for the panic (#71), but to let you know the panic was caused by a workaround put in place due to a lack of healthchecking in the SD libs for prometheus ([details here](https://github.com/jacksontj/promxy/blob/master/servergroup/servergroup.go#L126)  -- TLDR without healthchecks if a downstream prom dies it waits indefinitely to connect to it until it drops out of SD).

**Comment by jacksontj** [comment]
I have also created #72 to work on a long-term fix to the underlying issue, but this panic should be taken care of now. If you see any other issues please let me know! Thanks again for your help!
